src/main/java/core/basesyntax/strategy/TransactionStrategyImpl.java

```
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.transaction.TransactionHandler;
+import java.util.Map;
+
+public class TransactionStrategyImpl implements TransactionStrategy {
+    private Map<FruitTransaction.Operation, TransactionHandler> transactionHandlerMap;
+
+    public TransactionStrategyImpl(Map<FruitTransaction.Operation,
+            TransactionHandler> transactionHandlerMap) {
+        this.transactionHandlerMap = transactionHandlerMap;
+    }
+
+    @Override
+    public TransactionHandler get(FruitTransaction.Operation operation) {
+        return transactionHandlerMap.get(operation);
```

Consider handling the case where the `operation` is not present in the `transactionHandlerMap`. You could throw an exception or return a default handler to avoid potential `NullPointerException` when an operation is not mapped.